Source/Editor/Gizmo/TransformGizmoBase.Draw.cs

```diff
@@ -6,206 +6,260 @@ namespace FlaxEditor.Gizmo
+                            dir = WorldTransform.Right;
+                            DebugDraw.DrawLine(Vector3.Right * WorldTransform.Orientation * 100000f + WorldTransform.Translation, Vector3.Left * WorldTransform.Orientation * 100000f + WorldTransform.Translation, Resources.XAxisColor, 0f, true);
+                            break;
+                        case Axis.Y:
+                            dir = WorldTransform.Up;
+                            DebugDraw.DrawLine(Vector3.Up * WorldTransform.Orientation * 100000f + WorldTransform.Translation, Vector3.Down * WorldTransform.Orientation * 100000f + WorldTransform.Translation, Resources.YAxisColor, 0f, true);
+                            break;
+                        case Axis.XY:
+                            DebugDraw.DrawLine(Vector3.Right * WorldTransform.Orientation * 100000f + WorldTransform.Translation, Vector3.Left * WorldTransform.Orientation * 100000f + WorldTransform.Translation, Resources.XAxisColor, 0f, true);
+                            DebugDraw.DrawLine(Vector3.Up * WorldTransform.Orientation * 100000f + WorldTransform.Translation, Vector3.Down * WorldTransform.Orientation * 100000f + WorldTransform.Translation, Resources.YAxisColor, 0f, true);
+                            break;
+                        case Axis.Z:
+                            dir = WorldTransform.Forward;
+                            DebugDraw.DrawLine(Vector3.Forward * WorldTransform.Orientation * 100000f + WorldTransform.Translation, Vector3.Backward * WorldTransform.Orientation * 100000f + WorldTransform.Translation, Resources.ZAxisColor, 0f, true);
+                            break;
+                        case Axis.ZX:
+                            DebugDraw.DrawLine(Vector3.Right * WorldTransform.Orientation * 100000f + WorldTransform.Translation, Vector3.Left * WorldTransform.Orientation * 100000f + WorldTransform.Translation, Resources.XAxisColor, 0f, true);
+                            DebugDraw.DrawLine(Vector3.Forward * WorldTransform.Orientation * 100000f + WorldTransform.Translation, Vector3.Backward * WorldTransform.Orientation * 100000f + WorldTransform.Translation, Resources.ZAxisColor, 0f, true);
+                            break;
+                        case Axis.YZ:
+                            DebugDraw.DrawLine(Vector3.Forward * WorldTransform.Orientation * 100000f + WorldTransform.Translation, Vector3.Backward * WorldTransform.Orientation * 100000f + WorldTransform.Translation, Resources.YAxisColor, 0f, true);
+                            DebugDraw.DrawLine(Vector3.Up * WorldTransform.Orientation * 100000f + WorldTransform.Translation, Vector3.Down * WorldTransform.Orientation * 100000f + WorldTransform.Translation, Resources.ZAxisColor, 0f, true);
+                            break;
+                    }
+                    bool flag2 = _activeMode == Mode.Rotate;
+                    if (flag2)
+                    {
+                        bool flag3 = !dir.IsZero;
```

Comment:
Aw, that's where the Git thingy, followed by decompiling your own code happened, right?

Reply:
yes